### PR TITLE
docs(api): Updates API changlog

### LINKED ISSUE #4678: Release 4.1 of API and v2 of webhooks
Documentation

Document all API changes

### PR DESCRIPTION
**Pull request by ERosendo**
This PR closes #4678 by updating the REST API changelog template to reflect all changes made during the recent sprint cycle.

image

Note: This PR should be merged once all sub-issues related to #4678 are resolved.

### CODE REVIEW

**Review by mlissner — CHANGES_REQUESTED** [review]
I think we can do a bit more here:

 - Include links to the PRs and issues. 
 - Update the API pages to say v4.1 instead of v4.0.
 - Link to the documentation for each feature from the release notes.

I think it's also worth separating the many enhancements into a bulleted list or something. It's a bit tricky to realize that the various sentences are orthogonal to each other.

**Comment by ERosendo** [comment]
@mlissner I've applied your suggestions, and here's the updated v4.1 section:

image

File: cl/api/templates/rest-change-log.html [hunk 1]
```diff
@@ -30,6 +30,31 @@ <h4 class="v-offset-below-3">
...
+          (<a href="https://github.com/freelawproject/courtlistener/issues/4312">Issue #4312</a>, <a
+            href="https://github.com/freelawproject/courtlistener/pull/4712">PR #4712</a>).
+        </li>
+        <li><strong>Flexible Docket Entry Sorting:</strong> The <code>order_by</code> query parameter has been added to the <a href="{% url "pacer_api_help" %}#docket-entry-endpoint">Docket Entry endpoint</a>, enabling you to sort results based on various fields.
+          (<a href="https://github.com/freelawproject/courtlistener/issues/4061">Issue #4061</a>, <a
+            href="https://github.com/freelawproject/courtlistener/pull/4700">PR #4700</a>).
+        </li>
+        <li><strong>Deeper Filtering for Parties and Attorneys:</strong> The <a href="{% url "pacer_api_help" %}#party-endpoint">parties</a> and <a href="{% url "pacer_api_help" %}#attorney-endpoint">attorneys</a> API endpoint has been upgraded with the filter_nested_results=True query parameter. This enables granular filtering of nested data, providing more control over the retrieved information.
```
Comment (mlissner) [inline cl/api/templates/rest-change-log.html:48]:
```suggestion
        <li><strong>Deeper Filtering for Parties and Attorneys:</strong> The <a href="{% url "pacer_api_help" %}#party-endpoint">parties</a> and <a href="{% url "pacer_api_help" %}#attorney-endpoint">attorneys</a> API endpoint has been upgraded with the <code>filter_nested_results=True</code> query parameter. This enables granular filtering of nested data, providing more control over the retrieved information.
```

File: cl/api/templates/rest-docs-vlatest.html [hunk 2]
```diff
@@ -24,7 +24,7 @@ <h4 class="v-offset-below-2">
     {% include "includes/toc_sidebar.html" %}
   </div>
   <div class="col-xs-12 col-md-8 col-lg-6">
-    <h1 id="about">REST API &ndash; v4.0</h1>
+    <h1 id="about">REST API &ndash; v4.1</h1>
```
Comment (mlissner) [inline cl/api/templates/rest-docs-vlatest.html:27]: Let's catch this in the HTML title too.

**Review by mlissner — APPROVED** [review]
Looks close. 

The ordering comment isn't quite right though. The `order_by` parameter wasn't added, just additional fields to use with it: 

https://github.com/freelawproject/courtlistener/pull/4700/files#diff-a77be2c717f6c4b90d77faee73d82d017d6d7fbf2637d3e6d3d0b8891f11356a

A couple little tweaks too. It's important to get this right so that we can advertise all the little enhancements.

Thank you!

File: cl/api/templates/rest-docs-vlatest.html (hunk 2 above)
Comment (ERosendo) [reply cl/api/templates/rest-docs-vlatest.html:27]: Thanks for bringing this to my attention.

File: cl/api/templates/rest-change-log.html (hunk 1 above)
Comment (ERosendo) [reply cl/api/templates/rest-change-log.html:48]: Thanks!

**Comment by ERosendo** [comment]
@mlissner Thanks for your feedback. I've implemented the suggested changes and refined the docket entry endpoint's ordering comment.

**Review by mlissner — APPROVED** [review]

**Comment by mlissner** [comment]
Great. This is ready to go. Just need to merge the filtering one first.

**Pull request merged**